### ISSUE TITLE
Confusion with train, dev and test datasets

### ISSUE THREAD

**Issue by sfpoveda** [seisbench/seisbench#250]
Hi,

Im a little bit confused about the use given to each dataset in the tutorial notebook of "03a_training_phasenet.ipynb". I understand the train dataset is used to train the model, that is the data which the model uses to learn from. But i am not really what is the function of the dev dataset, does it infleunces the training? because as I understand the code of the notebook, it is only giving a measure of the performance of the model with epochs. And if that is true, what is the test set for? Isn' the dev dataset already doing what the test set is meant for?

**Comment by yetinam** [comment]
Hi @sfpoveda ,

the train/dev/test splits have the following purpose:

- The model is trained on the "train" dataset, i.e., the model parameters are optimized on this dataset.
- The "dev" dataset is used for optimizing so-called hyperparameters, e.g., the learning rate, the batch size, the number of epochs, but also training schemes and model architecture.
- The "test" dataset is used to measure the final performance. It should only be touched once all the hyperparameters have been selected.

Tuning the hyperparameters on the test set is considered bad practice, as this might lead to hyperparameter overfitting, i.e., overly optimistic estimates of the model performance.

As in the example you mentioned we do not tune hyperparamters, we are not actually making use of all three subsets.

I hope this helped answering your question.

**Comment by sfpoveda** [comment]
Thanks, it is clear now!

**Comment by yetinam** [comment]
The example is using the train part for training and the dev part to measure performance. The test is not used here. However, this is just a simple example, not a comprehensive workflow. In a full workflow, you'd use the sets as described above. The benchmark repository is an example of that.
